### PR TITLE
Carve pilot kills database connections after prompting

### LINKED ISSUE #35: Carve Pilot from database not working
I attempted to use this command on my test system and got an unhelpful failure message: 

>MessageBox >> Something has gone wrong. Check output and correct any issues.

Not sure what output to check. The log file showed the same message as above with no additional info.

### PR DESCRIPTION
**Pull request by elipup**
Added some helper functions to revoke and grant connect privs and kill connections using pg_terminate_backend().
Fixes #35

### CODE REVIEW

File: database.sh [hunk 1]
```diff
@@ -332,13 +332,21 @@ carve_pilot() {
...
-    if [ $RET -eq 1 ]; then
-        msgbox "Something has gone wrong. Check output and correct any issues."
-        do_exit
-    else
-        msgbox "Database "$PILOT" has been created"
+    if (whiptail --title "Warning" --yesno "This will kill all active connections to the database, if any.  Continue?" 10 60) then
+        remove_connect_priv $SOURCE
+        kill_database_connections $SOURCE
+
+        sudo su - postgres -c "psql postgres -q -h $PGHOST -p $PGPORT -c \"CREATE DATABASE \"$PILOT\" TEMPLATE \"$SOURCE\" OWNER admin;\""
```
Comment (davidbeauchamp) [inline database.sh]: log_exec here please

File: database.sh [hunk 2]
```diff
@@ -332,13 +332,21 @@ carve_pilot() {
...
-        do_exit
-    else
-        msgbox "Database "$PILOT" has been created"
+    if (whiptail --title "Warning" --yesno "This will kill all active connections to the database, if any.  Continue?" 10 60) then
+        remove_connect_priv $SOURCE
+        kill_database_connections $SOURCE
+
+        log_exec psql postgres -U postgres -q -h $PGHOST -p $PGPORT -c "CREATE DATABASE "$PILOT" TEMPLATE "$SOURCE" OWNER admin;"
+        RET=$?
+        if [ $RET -eq 1 ]; then
```
Comment (gilmoskowitz) [inline database.sh]: Why not use `if log_exec createdb ... ; then ...` instead of `$RET` and `psql`?

**Comment by elipup** [comment]
I was following the current style.  I think it would be better to do style changes all at once in a separate PR.

**Comment by davidbeauchamp** [comment]
I originally chose to do everything with psql because I could be sure it existed everywhere xtuple-admin-utility would be run.

**Comment by gilmoskowitz** [comment]
There are two separate questions:

- Why `psql` instead of `createdb`? You've answered that one, @davidbeauchamp 
- Why `command ; RET=$? ; ... `? @elipup answered for this PR but not for the general case

**Comment by davidbeauchamp** [comment]
I went with that particular style because I had run into instances where $? changed before I was able to compare it, so I stuck it into a variable and checked that instead. I guess the alternative is to just use || or && as part of the command..

**Pull request merged**